main/FatesHistoryInterfaceMod.F90

```diff
@@ -5240,6 +5252,160 @@ end subroutine update_history_hifrq_sitelevel
+         foundbaregroundpatch = .false.
+         cpatch => sites(s)%oldest_patch
+         do while(associated(cpatch))
+            if (cpatch%land_use_label .eq. nocomp_bareground_land .and. .not. foundbaregroundpatch) then
+               foundbaregroundpatch = .true.
+               do i_lu = 1, n_landuse_cats
+                  if ( landuse_statevector(i_lu) .gt. rsnbl_math_prec ) then
+                     hio_tsa_si_landuse(io_si,i_lu) = hio_tsa_si_landuse(io_si,i_lu) + &
+                          bc_in(s)%t2m_pa(cpatch%patchno) * &
+                          (landuse_statevector(i_lu) - canopy_area_bylanduse(i_lu)) / landuse_statevector(i_lu)
+
+                     hio_sw_abs_si_landuse(io_si,i_lu) = hio_sw_abs_si_landuse(io_si,i_lu) + &
+                          bc_in(s)%swabs_pa(cpatch%patchno) * &
+                          (landuse_statevector(i_lu) - canopy_area_bylanduse(i_lu)) / landuse_statevector(i_lu)
+
+                     hio_lw_net_si_landuse(io_si,i_lu) = hio_lw_net_si_landuse(io_si,i_lu) + &
+                          bc_in(s)%netlw_pa(cpatch%patchno) * &
+                          (landuse_statevector(i_lu) - canopy_area_bylanduse(i_lu)) / landuse_statevector(i_lu)
+
+                     hio_shflux_si_landuse(io_si,i_lu) = hio_shflux_si_landuse(io_si,i_lu) + &
+                          bc_in(s)%shflux_pa(cpatch%patchno) * &
+                          (landuse_statevector(i_lu) - canopy_area_bylanduse(i_lu)) / landuse_statevector(i_lu)
+
+                     hio_lhflux_si_landuse(io_si,i_lu) = hio_lhflux_si_landuse(io_si,i_lu) + &
+                          bc_in(s)%lhflux_pa(cpatch%patchno) * &
+                          (landuse_statevector(i_lu) - canopy_area_bylanduse(i_lu)) / landuse_statevector(i_lu)
+                  end if
+               end do
+            end if
+            cpatch => cpatch%younger
+         end do
+
+         ! instead of leaving the values for unoccupied areas as zero, set as missing values
+         do i_lu = 1, n_landuse_cats
+
+            ! if a given land use type is not present, set the value as missing
+            if ( landuse_statevector(i_lu) .le. rsnbl_math_prec ) then
+               hio_tsa_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+               hio_sw_abs_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+               hio_lw_net_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+               hio_shflux_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+               hio_lhflux_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+            end if
+
+            ! for tveg, ignore if there is no vegetation present on any patches of a given land use type
+            if ( canopy_area_bylanduse(i_lu) .le. rsnbl_math_prec ) then
+               hio_tveg_si_landuse(io_si,i_lu) = hlm_hio_ignore_val
+            end if
+
+         end do
+
+
+         cpatch => sites(s)%oldest_patch
+         do while(associated(cpatch))
```

Comment:
Why is this `do` loop separate from the others? Is it because GPP is the only one with the `ccohort%n * dt_tstep_inv` normalization factor? A comment would be useful to explain this.